_posts/2016-04-26-distributed-game-of-life-in-elixir.md

```diff
@@ -0,0 +1,1065 @@
+---
+layout: post
+title:  "Distributed Game of Life in Elixir"
+date:   2016-04-26 22:34:00 +0100
+author: "Artur Trzop"
+categories: elixir
+tags: elixir distributed game
+---
+
+I wrote my first game in Elixir. It's a popular thing - Conway's Game of Life - but it gets quite interesting when you solve it in a functional language, especially when you can see how [actor model](https://en.wikipedia.org/wiki/Actor_model) works and how actors are distributed across servers in network.
+
+<img src="/images/blog/posts/distributed-game-of-life-in-elixir/game_of_life_logo.png" style="float:right;margin-left:20px;" />
+
+__What I'm going to show you is:__
+
+* How to write rules for game of life with tests in Elixir
+* Parallel tasks across lightweight processes (actors) so we utilize all CPU cores
+* How to distribute work across nodes so the game can be executed by many servers in the cluster
+* How to use GenServer behaviour, TaskSupervisor and Agents in Elixir
+
+This project and the full [source code can be found here](https://github.com/BeyondScheme/elixir-game_of_life).
+
+# Demo
+
+Let's start with watching quick demo how the game works.
+
+<script type="text/javascript" src="https://asciinema.org/a/44233.js" id="asciicast-44233" async data-preload="true"></script>
+
+As you see we started node1 where the game is running and board is printed on the screen. We started also the second node and connected it to the first one. From the second node, we added new cells to the board. Both nodes are responsible for processing the game but only the first node is a master with info about the current state of the game. We can connect more nodes to the cluster so game processing can happen on all of the nodes. You are going to learn in this article how to make it happen.
```

Comment:
- "As you can see, node1 represents running game and board on the screen".
- "The second node was also started and connected to the first one".
- "Both nodes are responsible for processing the game," - comma after game
- "master with info" - with information